src/main/java/org/elasticsearch/search/aggregations/pipeline/movavg/MovAvgPipelineAggregator.java

```diff
@@ -158,13 +157,35 @@ public InternalAggregation reduce(InternalAggregation aggregation, ReduceContext
 
             double[] predictions = model.predict(values, predict);
             for (int i = 0; i < predictions.length; i++) {
```

Comment:
why not start `i` from 1 and have the end condition as `i <= predictions.length`? You seem to always use `i+1` anyway?